From 493fd65c116f423e9bb0a374314775fd475575c3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20K=2E=20H=C3=BCttel?= <dilfridge@gentoo.org>
Date: Tue, 29 Aug 2017 22:58:50 +0200
Subject: [PATCH 04/12] Revert "	* sysdeps/posix/getaddrinfo.c (gaih_inet):
 Only use gethostbyname4_r"

This reverts commit 8479f23aa1d5e5477a37f46823856bdafaedfa46.

See for the background
  https://sourceware.org/bugzilla/show_bug.cgi?id=16826
  https://bugs.gentoo.org/show_bug.cgi?id=600632

This makes test resolv/tst-res_use_inet6 fail. Disable it.
(cherry picked from commit 06a2a073a2c6fb3e1489af96e125daca248d0a03)
---
 sysdeps/posix/getaddrinfo.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/sysdeps/posix/getaddrinfo.c b/sysdeps/posix/getaddrinfo.c
index ed04e564f9..ce9e0d342d 100644
--- a/sysdeps/posix/getaddrinfo.c
+++ b/sysdeps/posix/getaddrinfo.c
@@ -739,13 +739,8 @@ gaih_inet (const char *name, const struct gaih_service *service,
 	  while (!no_more)
 	    {
 	      no_data = 0;
-	      nss_gethostbyname4_r *fct4 = NULL;
-
-	      /* gethostbyname4_r sends out parallel A and AAAA queries and
-		 is thus only suitable for PF_UNSPEC.  */
-	      if (req->ai_family == PF_UNSPEC)
-		fct4 = __nss_lookup_function (nip, "gethostbyname4_r");
-
+	      nss_gethostbyname4_r *fct4
+		= __nss_lookup_function (nip, "gethostbyname4_r");
 	      if (fct4 != NULL)
 		{
 		  while (1)
-- 
2.26.2

